### PR TITLE
docs: Update hotplug.md

### LINKED ISSUE #1563: Update hotplug.md
Needs details of device hotplug.

### PR DESCRIPTION
**Pull request by sboeuf**
Update the hotplug documentation to cover PCI devices and `virtio-mem`.

Fixes #1563

### CODE REVIEW

File: docs/hotplug.md [hunk 1]
```diff
@@ -111,3 +115,134 @@ Due to guest OS limitations is is necessary to ensure that amount of memory adde
...
+	--net "tap=,mac=,ip=,mask=" \
+    --api-socket=/tmp/ch-socket
+$ popd
+```
+
+To ask the VMM to expand the RAM for the VM (request is in bytes):
+
+```shell
+curl -H "Accept: application/json" -H "Content-Type: application/json" -i -XPUT --unix-socket /tmp/ch-socket -d "{ \"desired_ram\" : 3221225472}" http://localhost/api/v1/vm.resize
+```
```
Comment (rbradford) [inline docs/hotplug.md:145]: Most users will use ch-remote instead.

**Review by rbradford — COMMENTED** [review]
Looks like the whole file needs to be updated to reflect `ch-remote`. I guess it predates its availability

File: docs/hotplug.md (hunk 1 above)
Comment (sboeuf) [reply docs/hotplug.md:145]: Yes I'll add one more commit to replace everything with `ch-remote`.

**Comment by sboeuf** [comment]
@rbradford PR updated!

File: docs/hotplug.md [hunk 2]
```diff
@@ -111,3 +115,134 @@ Due to guest OS limitations is is necessary to ensure that amount of memory adde
...
+```shell
+$ pushd $CLOUDH
+$ sudo setcap cap_net_admin+ep ./cloud-hypervisor/target/release/cloud-hypervisor
+$ ./cloud-hypervisor/target/release/cloud-hypervisor \
+	--kernel custom-vmlinux.bin \
+	--cmdline "console=ttyS0 console=hvc0 root=/dev/vda1 rw" \
+	--disk path=focal-server-cloudimg-amd64.raw \
+	--memory size=1024M,hotplug_size=8192M,hotplug_method=virtio-mem \
+	--net "tap=,mac=,ip=,mask=" \
+    --api-socket=/tmp/ch-socket
```
Comment (rbradford) [inline docs/hotplug.md:137]: nit: Something weird is going on with your indentation?

File: docs/hotplug.md [hunk 3]
```diff
@@ -111,3 +115,134 @@ Due to guest OS limitations is is necessary to ensure that amount of memory adde
...
+```shell
+$ sudo setcap cap_net_admin+ep ./cloud-hypervisor/target/release/cloud-hypervisor
+$ ./cloud-hypervisor/target/release/cloud-hypervisor \
+	--kernel custom-vmlinux.bin \
+	--cmdline "console=ttyS0 console=hvc0 root=/dev/vda1 rw" \
+	--disk path=focal-server-cloudimg-amd64.raw \
+	--cpus boot=4 \
+	--memory size=1024M \
+	--net "tap=,mac=,ip=,mask=" \
+    --api-socket=/tmp/ch-socket
```
Comment (rbradford) [inline docs/hotplug.md:173]: nit: Ditto?

**Review by rbradford — APPROVED** [review]

File: docs/hotplug.md (hunk 2 above)
Comment (sboeuf) [reply docs/hotplug.md:137]: Oh weird, let me fix that :)

**Review by likebreath — APPROVED** [review]

**Pull request merged**